runtime/tchannel_client.go

```diff
@@ -193,6 +193,10 @@ func (c *TChannelClient) call(
 	if c.routingKey != nil {
 		ctxBuilder.SetRoutingKey(*c.routingKey)
 	}
+	rd, ok := ctx.Value("rd").(string)
```

Comment:
You should expose a function like `WithRoutingDelegate(Context, string) Context` so that the implementation (i.e. string `"rd"`) is hidden to callers and more semantic.

Reply:
Good point, it's done now.